port/unix/omrsysinfo.c

```diff
@@ -6495,140 +6490,159 @@ readCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFl
+
+					if (2 != rc) {
+						Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
+						rc = portLibrary->error_set_last_error_with_message_format(
+								portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED,
+								"sscanf, unexpected format of %s file, expected format = %s, return code = %d",
+								OMR_PROC_PID_ONE_CGROUP_FILE, PROC_PID_CGROUP_SYSTEMD_ENTRY_FORMAT, rc);
+						goto _end;
+					}
+				} else if (3 != rc) {
 					Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
-					rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
+					rc = portLibrary->error_set_last_error_with_message_format(
+							portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED,
+							"sscanf, unexpected format of %s file, expected format = %s, return code = %d",
+							OMR_PROC_PID_ONE_CGROUP_FILE, PROC_PID_CGROUPV1_ENTRY_FORMAT, rc);
 					goto _end;
 				}
-			} else if (3 != rc) {
-				Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
-				rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
-				goto _end;
-			}
 
-			if ((0 != strcmp(ROOT_CGROUP, cgroup)) && (0 != strcmp(SYSTEMD_INIT_CGROUP, cgroup))) {
-				*inContainer = TRUE;
-				break;
+				if ((0 != strcmp(ROOT_CGROUP, cgroup)) && (0 != strcmp(SYSTEMD_INIT_CGROUP, cgroup))) {
+					inContainer = TRUE;
+					break;
+				}
 			}
+			rc = 0;
 		}
-		rc = 0;
-	}
-
-	/* Read the first line in /proc/1/sched if it exists.
-	 * Outside a container, the initialization process is either "init" or "systemd".
-	 * A containerized environment can be assumed for any other initialization process.
-	 */
-	if ((0 == rc) && (!*inContainer) && (0 == access(OMR_PROC_PID_ONE_SCHED_FILE, F_OK))) {
-		FILE *schedFile = fopen(OMR_PROC_PID_ONE_SCHED_FILE, "r");
-		char buffer[PATH_MAX];
 
-		if (NULL == schedFile) {
-			int32_t osErrCode = errno;
-			Trc_PRT_isRunningInContainer_fopen_failed(OMR_PROC_PID_ONE_SCHED_FILE, osErrCode);
-			rc = portLibrary->error_set_last_error(
-					portLibrary,
-					osErrCode,
-					OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_FOPEN_FAILED);
-			goto _end;
-		}
+		/* Read the first line in /proc/1/sched if it exists.
+		 * Outside a container, the initialization process is either "init" or "systemd".
+		 * A containerized environment can be assumed for any other initialization process.
+		 */
+		if ((0 == rc) && (!inContainer) && (0 == access(OMR_PROC_PID_ONE_SCHED_FILE, F_OK))) {
+			FILE *schedFile = NULL;
+			char buffer[PATH_MAX];
 
-		if (NULL == fgets(buffer, sizeof(buffer), schedFile)) {
-			if (0 != ferror(schedFile)) {
+			if (NULL == (schedFile = fopen(OMR_PROC_PID_ONE_SCHED_FILE, "r"))) {
```

Comment:
Again, don't move the assignment into the `if` test.

Reply:
The assignment has been separated from the test.